subsys/bluetooth/host/scan.h

```diff
@@ -5,12 +5,95 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+#ifndef SUBSYS_BLUETOOTH_HOST_SCAN_H_
+#define SUBSYS_BLUETOOTH_HOST_SCAN_H_
+
+#include <stdint.h>
+
+#include <zephyr/sys/atomic.h>
+#include <zephyr/bluetooth/bluetooth.h>
+
+/**
+ * Reasons why a scanner can be running.
+ * Used as input to @ref bt_le_scan_set_flag_and_update
+ * and @ref bt_le_scan_clear_flag_and_update.
+ */
+enum bt_le_scan_enabled_flag {
+	/** The application explicitly instructed the stack to scan for advertisers
+	 * using the API @ref bt_le_scan_start().
+	 * May not be used explicitly.
+	 */
+	BT_LE_SCAN_ENABLED_EXPLICIT_SCAN,
+
+	/**
+	 * Syncing to a periodic advertiser.
+	 */
+	BT_LE_SCAN_ENABLED_SYNC_SYNCING,
+
+	/**
+	 * Scanning to find devices to connect to.
+	 */
+	BT_LE_SCAN_ENABLED_SCAN_BEFORE_INITIATE,
+
+	/**
+	 * Special state for a NOP for @ref bt_le_scan_update_and_reconfigure to not add/remove any
+	 * flag why the scanner should be running.
+	 */
+	BT_LE_SCAN_ENABLED_NONE,
+	BT_LE_SCAN_ENABLED_NUM_FLAGS,
+};
+
 void bt_scan_reset(void);
 
 bool bt_id_scan_random_addr_check(void);
+bool bt_le_scan_active_scanner_running(void);
 
 int bt_le_scan_set_enable(uint8_t enable);
 
 struct bt_le_per_adv_sync *bt_hci_get_per_adv_sync(uint16_t handle);
 
 void bt_periodic_sync_disable(void);
+
+/**
+ * Start / update the scanner.
+ *
+ * This API updates the reasons why the scanner is running.
+ * Multiple reasons can be enabled at the same time.
+ * Depending on all the reasons why the scanner is enabled, scan parameters are selected
+ * and the scanner is started or updated, if needed.
+ * This API may update the scan parameters, for example if the scanner is already running
+ * when another reason that demands higher duty-cycle is being added.
+ *
+ * Every SW module that informs the scanner that it should run, needs to eventually remove
+ * the flag again using @ref bt_le_scan_clear_flag_and_update once it does not require
+ * the scanner to run, anymore.
+ *
+ * @param flag reason why the scanner should be running
+ *
+ * @return 0 in case of success, or a negative error code on failure.
+ */
+int bt_le_scan_set_flag_and_update(enum bt_le_scan_enabled_flag flag);
```

Comment:
s/flag/reason maybe?

Reply:
I requested the change from reason -> flag: https://github.com/zephyrproject-rtos/zephyr/pull/73923#discussion_r1648634936

At least to me, it's not ideal nor clear that `reason` is a bitfield of reasons, where `flag` is more commonly known for that. 

In the end it's just naming, so not going to block it if we revert to `reason`.

Reply:
Naming is hard :D 

In the host we use `flags` for bitfields many places, such as the `bt_le_ext_adv.flags`, `bt_dev.flags`, `bt_att_chan.flags`, etc., which is why I though re-using that terminology here made sense. 

The function does 2 things: It sets a specific flag (in this case it seems to only ever set a single bit/flag since it uses `atomic_set_bit`), and then updates the scanning.